\documentclass[a4paper,12pt]{article}
\usepackage[utf8]{inputenc}
\usepackage{graphicx}
\usepackage[english]{babel}
\usepackage{geometry}
\usepackage{setspace}
\usepackage{minted}
\usepackage{booktabs}

\geometry{margin=1in}
\onehalfspacing

\begin{document}

\begin{titlepage}
    \begin{center}
        % adjust the logo here
        % Title
        {\Large \textbf{Genetic Data Analysis of Rheumatoid arthritis and Clouston's disease}} \\[1cm] 
        
        % Subtitle
        \textit{Module: M1 Méthodologie de traitement de données NGS} \\[3cm]
        

        \vfill % to push the authors to the bottom of the page

        \textbf{Authors} \\[0.5cm]
        \textsc{Ngoc VU, Denys BURYI}


    \end{center}
\end{titlepage}

\newpage


\section*{Introduction}

We have carried out linkage analyses (lod-score or affected sib-pairs) 
and association analyses (case-control or familial) for a monogenic disease, then for a multifactorial disease:
\begin{enumerate}
    \item Study of a monogenic disease: Clouston's disease
    \begin{enumerate}
        \item Genetic linkage analysis (lodscore)
        \item Familial association analysis
    \end{enumerate}
    \item Study of a multifactorial disease: rheumatoid arthritis
    \begin{enumerate}
        \item Genetic linkage analysis (affected sib-pairs)
        \item Genome-Wide case-control association analysis
    \end{enumerate}
\end{enumerate}

 
The data and programs were provided by our professor VALÉRIE CHAUDRU.

\section{Monogenic disease: Clouston's disease}

Clouston syndrome is mainly characterized by abnormal nails (thickened, slow-growing, brittle) 
and alopecia (hair loss). \textit{Palmoplantar hyperkeratosis} (commonly known as horniness) is 
also sometimes observed. 
This is a rare disease, with a prevalence ranging from 1 to 9 in 100,000. It is autosomal dominant. 
Penetrance of the disease is complete, but its expression is variable, even in patients from the same family.
Our aim in this part of the analysis was to:

\begin{itemize}
    \item Perform linkage analysis using the lodscore method to locate the Clouston disease gene 
    using genetic markers located on chromosome 13.
    \item Interpret data from a \texttt{.vcf} file containing genetic variants identified in a 
    case-control sample and perform association analysis for a candidate variant.
    \item Search for the causal genetic variant using familial association analysis.
\end{itemize}

\subsection{Genetic linkage analysis - lodscore method}

We have analyzed the data contained in the \texttt{fam.txt} file using \texttt{paramlink} package in R. 
First we performed some initial data exploration using \texttt{linkdat()} and \texttt{summary()} functions:

    
    \begin{minted}[frame=lines, linenos]{r}
        fam = read.table('fam.txt')
        x = linkdat(fam)
        summary(x)
        \end{minted}


\newpage
We found that the file contains data of a large family of 47 individuals, comprising 22 affected and
 25 unaffected members. These individuals are organized into 10 nuclear subfamilies and 
have been genotyped for 13 genetic markers on chromosome 13. For 11 individuals in the dataset no 
parent data was available (further referred to as "founders"). Figure \ref{fig:pedigree} shows 
the pedigree plot of the family.

\begin{figure}[ht!] % ht! means here at the top, despite placing rules 
    \centering
    \includegraphics[width=\textwidth]{images/pedigree_plot.png}
    \caption{Pedigree plot of the family along with their affection status and genotypes with 
    regards to marker 1.
    Affected - light green, Unaffected - white. Sex: square/circle - M/F. Genotype: alleles of the marker 1.}
    \label{fig:pedigree} % this allows to reference the figure with \ref{fig:pedigree}

    
\end{figure}

In addition to helping summarize the data, the \texttt{linkdat()} function also transforms the original data
and outputs a \texttt{linkdat} object that can be used to perform linkage analysis. \texttt{linkdat} object 
is a list containing among others, the following important components:
\begin{itemize}
    \item \texttt{pedigree} - data.frame with 5 columns (ID, FID, MID, SEX, AFF) describing the pedigree in linkage format.
    \item \texttt{orig.ids} - vector of original IDs of the individuals in the pedigree.
    \item \texttt{subnucs} - list of vectors of IDs of individuals in each nuclear family.
    \item \texttt{markerdata} - a list of \texttt{marker} objects describing information about the genetic markers.
\item \texttt{model} - a list of parameters for the linkage analysis.
\end{itemize}



\subsection{Mode of inheritance: Autosomal Dominant}

First, we have analyzed our data assuming an autosomal dominant mode of inheritance. Using the \texttt{setmodel()} function,
we have set the parameters to:
\begin{itemize}
    \item phenocopies = $10^{-5}$
    \item complete penetrance
    \item disease allele frequency = $10^{-5}$
\end{itemize}

We then performed the linkage analysis using the \texttt{lod()} function for a range of $\theta$ from 0 to 0.5 with a step of 0.05. 
We have saved the results in the \texttt{result\_dom} data frame for further analysis (see code below).

\begin{minted}[frame=lines, linenos]{r}
    xdom = setModel(x, model=1, penetrances = c(0.00001, 1, 1), dfreq = 0.00001)
    result_dom = lod(xdom, theta=c(0, 0.05, 0.1, 0.15, 0.2, 0.25, 0.3, 0.4, 0.5)) 
    result__dom_df = as.data.frame(result_dom)
\end{minted}

\begin{figure}[hb!] % b for bottom 
    \centering
    \includegraphics[width=\textwidth]{images/xdom_lod_analysis.png}
    \caption{LOD score analysis results for autosomal dominant mode of inheritance. Each column shows 
    Lod-score for a given marker, for a given $\theta$ value. Top row shows maximum Lod-score for each marker. 
    Conditional coloring: green to red - higest to lowest Lod-score.}   
    \label{fig:xdom_lodscore} 
    
\end{figure}

When performing LOD-score analysis, under the null hypothesis $\mathit{H_0}$ we assume 
that there is no linkage between the markers and the disease gene and $\mathit{\theta = 0.5}$. In other words if the two 
loci are unlinked, the probability of observing a recombinant event $\mathit{c = 50\%}$. Under an alternative
hypothesis $\mathit{H_1}$, if the two loci are indeed linked, we expect the probability of observing a recombinant 
event \textit{c} to fall within $0 < c < 50\%$. LOD-score is a logarithmic likelihood ratio:

\[
\mathit{Z(x) = \log_{10} \left[ \frac{L(c = x)}{L(c = 0.5)} \right]}
\]
A LOD score of 3 indicates a 1,000:1 likelihood that two genes are linked and inherited together with a given 
recombination rate(e.g. \textit{c=10\%}), compared to the likelihood under the null hypothesis.
For all markers except M13, the maximum LOD score exceeds 3 (see Figure \ref{fig:xdom_lodscore}).
These results suggest that the disease gene is likely in extremely close proximity to markers 1–10, 
and approximately 5 cM away from markers 11 and 12.

As for marker 13, it is highly probable that adding more families to the dataset would increase $\mathit{Z(max)}$ 
above 3. For now, the data suggest that the disease gene might be located approximately 15 cM from marker 13.

To validate these findings, we performed the same analysis using $\theta$ values ranging from 0 to 0.05 in 
increments of 0.01 (see Figure \ref{fig:xdom_lodscore_granular}).



\begin{figure}[t] % t for top 
    \centering
    \includegraphics[width=\textwidth]{images/xdom_lod_analysis_more_granular.png}
    \caption{LOD score analysis results for autosomal dominant mode of inheritance for more 
    granular $\theta$ values. Each column shows Lod-score for a given marker, for a given 
    $\theta$ value. Top row shows maximum Lod-score for each marker. Conditional coloring: 
    green to red - highest to lowest Lod-score.}   
    \label{fig:xdom_lodscore_granular} 
    
\end{figure}

Next, we calculated the confidence intervals for each marker. % explanation here of why Zmax - 1 
Visualizing the LOD score curve can be helpful for this purpose, so we plotted it for marker 1 as 
a representative example for markers 2–10 (Figure \ref{fig:lod_curve}). 
To determine the precise confidence intervals for the results, we ran the script located in 
the \texttt{paramlink.R} file. Results are shown in the table below.

\begin{figure}[ht] % t for top 
    \centering
    \includegraphics[width=\textwidth]{images/Lod-score_curve_M1.png}
    \caption{LOD-score curve for marker 1. LOD-score values on the y-axis, $\theta$ values on the x-axis. 
    Grey line represents the $\mathit{Z(max)-1}$ value.}  
    \label{fig:lod_curve} 
    
\end{figure}

\begin{table}[ht]
    \centering
    \begin{tabular}{ccc}
        \toprule
        \textbf{Marker} & \textbf{Lower} & \textbf{Upper} \\ 
        \midrule
        1 & 0 & 0.077 \\ 
        2 & 0 & 0.082 \\ 
        3 & 0 & 0.109 \\ 
        4 & 0 & 0.074 \\ 
        5 & 0 & 0.09 \\ 
        6 & 0 & 0.116 \\ 
        7 & 0 & 0.09 \\ 
        8 & 0 & 0.097 \\ 
        9 & 0 & 0.122 \\ 
        10 & 0 & 0.155 \\ 
        11 & 0.059 & 0.201 \\ 
        12 & 0.059 & 0.201 \\ 
        13 & NA & NA \\ 
        \bottomrule
    \end{tabular}
    \caption{Confidence intervals for each marker, with lower and upper bounds. For marker 13, results are inconclusive.}
    \label{tab:ci_bounds}
\end{table}


\newpage
    \section{Multifactorial disease: Rheumatoid Arthritis}

\end{document}